### ISSUE TITLE
laravel/framework 5.1.9 breaks search+where+paginate

### ISSUE THREAD

**Issue by smb** [jarektkaczyk/eloquence#22]
i have some issues when using search a where clause and laravels paginate (no issue when removing one of them).

works in 5.1.8, broken in 5.1.9.
```
$results = Aura::search(["*" . $query . "*"], true)->isLive()->sortable()->paginate(20);
```

works in 5.1.8 + 5.1.9:
```
$results = Aura::search(["*" . $query . "*"], true)->sortable()->paginate(20);
```

scopeIsLive:
```
return $query->where('auras.status','=', Config::get('constant.aura.status.online'));
```

caused by 
https://github.com/laravel/framework/commit/4c4891280c944bdf151eb9a39ab39bb1017bbd45

it removes the search bindings for the paginate query, leading to:

```
Next exception 'Illuminate\Database\QueryException' with message 'SQLSTATE[HY093]: Invalid parameter number (SQL: select count(*) as aggregate from (select `auras`.*, max(case when `auras`.`title` = 2 then 300 else 0 end + case when `auras`.`title` like ? then 100 else 0 end + case when `auras`.`title` like ? then 20 else 0 end + case when `auras`.`description` = ? then 150 else 0 end + case when `auras`.`description` like ? then 50 else 0 end + case when `auras`.`description` like ? then 10 else 0 end + case when `categories`.`title` = ? then 75 else 0 end + case when `categories`.`title` like ? then 25 else 0 end + case when `categories`.`title` like ? then 5 else 0 end) as relevance from `auras` left join `aura_categories` on `aura_categories`.`aura_id` = `auras`.`id` left join `categories` on `aura_categories`.`category_id` = `categories`.`id` where (`auras`.`title` like ? or `auras`.`description` like ? or `categories`.`title` like ?) group by `auras`.`id`) as `auras` where `relevance` >= 8.75 and `auras`.`status` = ?)' 
```

is there a way to work around this issue or should i adress this at laravel/framework?

**Comment by jarektkaczyk** [comment]
Meh, recent L5 updates bring one problem after another.. Many *fixes* are focused on single thing and break somewhere else.

Anyway, eloquence does some magic with `wheres` in order to improve performance, so I need to have a deeper look with 5.1.9. Pls be patient

**Comment by mydnic** [comment]
It seems that even when using only search + paginate, it doesn't work..

Fresh install of L5.1.16, I use this code in my Controller : 

```php
User::search('my@email.com')->paginate(1); // just for testing
```

This returns : 
```
SQLSTATE[HY000]: General error: 2031 (SQL: select count(*) as aggregate from (select `users`.*, max(case when `users`.`email` = ? then 15 else 0 end + case when `users`.`email` like ? then 5 else 0 end + case when `users`.`email` like ? then 1 else 0 end) as relevance from `users` where (`users`.`email` like ?) group by `users`.`id`) as `users` where `users`.`deleted_at` is null and `relevance` >= 0.25)
```


The full Model : 

```php
<?php

namespace App;

use Illuminate\Auth\Authenticatable;
use Illuminate\Auth\Passwords\CanResetPassword;
use Illuminate\Contracts\Auth\Access\Authorizable as AuthorizableContract;
use Illuminate\Contracts\Auth\Authenticatable as AuthenticatableContract;
use Illuminate\Contracts\Auth\CanResetPassword as CanResetPasswordContract;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Foundation\Auth\Access\Authorizable;
use Sofa\Eloquence\Eloquence;

class User extends Model implements AuthenticatableContract,
                                    AuthorizableContract,
                                    CanResetPasswordContract
{
    use Authenticatable, Authorizable, CanResetPassword, SoftDeletes, Eloquence;

    protected $searchableColumns = ['email'];

    /**
     * The database table used by the model.
     *
     * @var string
     */
    protected $table = 'users';

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = ['name', 'email', 'password'];

    /**
     * The attributes excluded from the model's JSON form.
     *
     * @var array
     */
    protected $hidden = ['password', 'remember_token'];
}
```

**Comment by levelel** [comment]
@mydnic I am stuck at 5.1.8 due to this issue...

**Comment by jarektkaczyk** [comment]
Fixed in `5.1.1`
